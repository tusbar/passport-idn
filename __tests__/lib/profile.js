const {readFileSync} = require('fs')

const {parse} = require('../../lib/profile')

describe('lib.profile', () => {
  describe('parse', () => {
    it('should parse profile', () => {
      const data = readFileSync('__tests__/__fixtures__/example.json', 'utf8')

      expect(parse(data)).toMatchSnapshot()
    })

    it('should parse profile with null email', () => {
      const data = readFileSync('__tests__/__fixtures__/example-null-email.json', 'utf8')

      expect(parse(data)).toMatchSnapshot()
    })
  })
})
