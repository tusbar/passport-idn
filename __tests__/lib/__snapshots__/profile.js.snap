// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.profile parse should parse profile 1`] = `
Object {
  "emails": Array [
    Object {
      "value": "mbl@somewhere.net",
    },
  ],
  "id": "mbl",
  "name": "M LE-JAMBON Paul",
  "username": "mbl",
}
`;

exports[`lib.profile parse should parse profile with null email 1`] = `
Object {
  "id": "mbl",
  "username": "mbl",
}
`;
